### PR TITLE
test: organize integration tests following js integration tests

### PR DESCRIPTION
**Pull request by malandis**
# tldr
In order for developers to easily compare integration test coverage vs
the JS suite, a package with known well coverage, we organize the
integration tests to mirror those.

# Changes
Specifically we chose filenames that are similar to those used in the
JS integration test repos. Then, in order to parallel the `describe`
blocks, we introduce a `mod` for each describe. The individual tests
therein should be parallel to the `it` calls in the JS repo.

We've organized the data plane commands into the following
files:
- cache_scalar.rs: commands where the value type is scalar
- cache_sorted_set.rs: commands where the value type is sorted set
- cache_key_exists.rs: commands to test existence of a cache key
- cache_item.rs: commands that apply to a cache item irrespective of
type.

Previously the tests under `cache_item.rs` had been grouped under
`cache_scalar.rs`. This wasn't as cohesive as the scalar tests group
by commands that operate on scalar-valued items whereas the item tests
apply to all cache keys.

In the future we ought to group the cache tests under a folder `cache`
and rename the integration tests as appropriate.

Note this PR did not add or delete tests; nor did it modify test behavior.

# Benefits
This also allows developers to easily run a subset of the
tests. A developer can run `cargo t <modname>` to just run the tests
therein. There is also IDE support for running all tests in a mod (see
below). Additionally, this allows us to shorten test function names
and use the same test function name for common
concerns ("nonexistent_cache").

# Demonstrations
Can now group by functionality and collapse as in the JS tests. This 
makes it easy to see at a glance which groups of tests/functionality we 
have not implemented:
image
We can proactively stub out mods based on the JS repo coverage.

Can now run a subset of tests by functionality in IDE:
image

Or on the command line:
image

And we get shorter test names:
image

### CODE REVIEW

**Review by anitarua — APPROVED** [review]

**Review by cprice404 — APPROVED** [review]
Might be nice to add a few of your comments from the PR description into the CONTRIBUTING.md.

Other than that looks great!
ship

**Comment by malandis** [comment]
> Might be nice to add a few of your comments from the PR description into the CONTRIBUTING.md.
> 
> Other than that looks great! ![ship]

Will do
